traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go

```diff
@@ -567,6 +582,22 @@ func (cg *TOCacheGroup) Update() (error, error, int) {
 	return nil, nil, http.StatusOK
 }
 
+func CheckIfExistsBeforeUpdate(tx *sqlx.Tx, cgID int) (error, *tc.TimeNoMod) {
```

Comment:
These functions look identical, except for the table in the select query.
Could they be refactored to use a common func, maybe something like 
```
// GetLastUpdated returns the last updated time of the given ID in the given table, whether the ID existed, and any error.
func GetLastUpdated(tx *sqlx.Tx, dbTable string, id int) (time.Time, bool, error)
```` 

In the `api` package?

Reply:
I have refactored it so that all the places that are NOT part of the CRUDer interfaces, call the common function in the `api` package, by passing in the table name like you suggested. 
For the `GenericUpdate` occurrences, we need it to be specific to each component, because we dont know what "type" is calling the function. Hence, each component that implements the CRUDer interface, has its own implementation of the above function.

Reply:
Could the function also take the type? So the CRUDer funcs all become something like:
```
func (v *TODeliveryServiceRequestComment) CheckIfExistsBeforeUpdate() (time.Time, error) {
  return GetLastUpdated(v.APIInfo().Tx, dbTable string, id int, v.Type())
}
```
?

Reply:
Absolutely 👍